src/equations/shallow_water_wet_dry_1d.jl

```diff
@@ -0,0 +1,677 @@
+                                                   orientation,
+                                                   Trixi.ShallowWaterEquations1D(equations.gravity,
+                                                                                 equations.H0,
+                                                                                 eps(),
+                                                                                 eps()))
+end
+
+# TODO: This function is currently exported by Trixi.jl. Needs to be uncommented when removed from Trixi.jl
+# """
+#     flux_nonconservative_chen_noelle(u_ll, u_rr,
+#                                      orientation::Integer,
+#                                      equations::ShallowWaterEquationsWetDry1D)
+
+# Non-symmetric two-point surface flux that discretizes the nonconservative (source) term.
+# The discretization uses the `hydrostatic_reconstruction_chen_noelle` on the conservative
+# variables.
+
+# Should be used together with [`FluxHydrostaticReconstruction`](@ref) and
+# [`hydrostatic_reconstruction_chen_noelle`](@ref) in the surface flux to ensure consistency.
+
+# Further details on the hydrostatic reconstruction and its motivation can be found in
+# - Guoxian Chen and Sebastian Noelle (2017)
+#   A new hydrostatic reconstruction scheme based on subcell reconstructions
+#   [DOI:10.1137/15M1053074](https://dx.doi.org/10.1137/15M1053074)
+# """
+# @inline function flux_nonconservative_chen_noelle(u_ll, u_rr,
+#                                                   orientation::Integer,
+#                                                   equations::ShallowWaterEquationsWetDry1D)
+
+#     # Pull the water height and bottom topography on the left
+#     h_ll, _, b_ll = u_ll
+#     h_rr, _, b_rr = u_rr
+
+#     H_ll = h_ll + b_ll
+#     H_rr = h_rr + b_rr
+
+#     b_star = min(max(b_ll, b_rr), min(H_ll, H_rr))
+
+#     # Create the hydrostatic reconstruction for the left solution state
+#     u_ll_star, _ = hydrostatic_reconstruction_chen_noelle(u_ll, u_rr, equations)
+
+#     # Copy the reconstructed water height for easier to read code
+#     h_ll_star = u_ll_star[1]
+
+#     z = zero(eltype(u_ll))
+#     # Includes two parts:
+#     #   (i)  Diagonal (consistent) term from the volume flux that uses `b_ll` to avoid
+#     #        cross-averaging across a discontinuous bottom topography
+#     #   (ii) True surface part that uses `h_ll` and `h_ll_star` to handle discontinuous bathymetry
+#     return SVector(z,
+#                    equations.gravity * h_ll * b_ll -
+#                    equations.gravity * (h_ll_star + h_ll) * (b_ll - b_star),
+#                    z)
+# end
+
+"""
+    flux_nonconservative_ersing_etal(u_ll, u_rr, orientation::Integer,
+                                     equations::ShallowWaterEquationsWetDry1D)
+
+!!! warning "Experimental code"
+    This numerical flux is experimental and may change in any future release.
```

Comment:
Will this still be experimental in the new package?

Reply:
No, I think we can remove the experimental tag. We tested this a lot in the meantime and I don't see why there would be any changes to it.